windows client daemon panic for pure SSO setup

**Describe the problem**
When the first time setting up windows client ui with SSO login, the client daemon panics silently, "Connect" button becomes broken, all fields in settings view are empty.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x9d6dd6]

goroutine 117 [running]:
github.com/netbirdio/netbird/client/server.(*Server).Login(0xc00010ca00, {0xfb7670, 0xc0001c41b0}, 0xc0002e05a0)
        /home/runner/work/netbird/netbird/client/server/server.go:239 +0xd16
github.com/netbirdio/netbird/client/proto._DaemonService_Login_Handler({0xda8dc0?, 0xc00010ca00}, {0xfb7670, 0xc0001c41b0}, 0xc0005ccd90, 0x0)
        /home/runner/work/netbird/netbird/client/proto/daemon_grpc.pb.go:159 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000c2000, {0xfbf818, 0xc000538b60}, 0xc0006b0d80, 0xc00008d200, 0x162c4a0, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.52.3/server.go:1336 +0xd33
google.golang.org/grpc.(*Server).handleStream(0xc0000c2000, {0xfbf818, 0xc000538b60}, 0xc0006b0d80, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.52.3/server.go:1704 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.52.3/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.52.3/server.go:963 +0x28a
```

**To Reproduce**
Steps to reproduce the behavior:
1. setup a server instance with SSO openid device login
2. install windows client via installer exe and launch the ui
3. right click the tray icon, click settings
4. update mgmt url and admin url, save
5. right click the tray icon, click connect
6. client daemon panics
7. if we execute `netbird service start` now, further operations have no problem

**Expected behavior**
the SSO login page should pop up

**NetBird status -d output:**
n/a

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
0.20.3

--- a/client/server/server.go
+++ b/client/server/server.go
@@ -236,7 +236,9 @@ func (s *Server) Login(callerCtx context.Context, msg *proto.LoginRequest) (*pro
 				}, nil
 			} else {
 				log.Warnf("canceling previous waiting execution")
-				s.oauthAuthFlow.waitCancel()
+				if s.oauthAuthFlow.waitCancel != nil {
+					s.oauthAuthFlow.waitCancel()
+				}
 			}
 		}
 

